src/model/Space.cpp

```diff
@@ -2197,6 +2198,77 @@ namespace model {
       return result;
     }
 
+    SpaceInfiltrationDesignFlowRate Space_Impl::setInfiltrationDesignFlowRateHelper() {
+      // We try to grab an existing one, whether that's on the SpaceType or Space, because we want to retain other parameters like Term Coefficients
+      auto spis = this->spaceInfiltrationDesignFlowRates();
+
+      boost::optional<SpaceInfiltrationDesignFlowRate> spi_;
+
+      // handle space type
+      if (auto sp_ = this->spaceType()) {
+        auto sp_spis = sp_->spaceInfiltrationDesignFlowRates();
+        if (!sp_spis.empty()) {
+          // if sp is used by multiple ParentObjects (includes Space, Building, excludes SpaceLoadInstances), create and use clone instead of original
+          if (sp_->getModelObjectSources<ParentObject>().size() > 1u) {
+            sp_ = sp_->clone().cast<SpaceType>();
+            sp_spis = sp_->spaceInfiltrationDesignFlowRates();
+            setSpaceType(*sp_);
+          }
+          // If the Space didn't have one, we move the SpaceType one to the space
+          if (spis.empty()) {
+            spi_ = std::move(sp_spis.front());
+            sp_spis.erase(sp_spis.begin());
+            spi_->setSpace(getObject<Space>());
+          }
+
+          for (auto& spi : sp_spis) {
+            spi.remove();
+          }
+        }
+      }
+
+      if (spi_) {
+        return *spi_;
+      }
+
+      if (spis.empty()) {
+        SpaceInfiltrationDesignFlowRate spi(this->model());
+        spi.setSpace(getObject<Space>());
+        return spi;
+      }
+
+      auto& spi = spis[0];
+      for (auto& spi : spis | std::views::drop(1)) {
+        spi.remove();
+      }
```

Comment:
First std::views spotted in OS SDK! yay!

Reply:
ugh. I can't keep up.

Reply:
At least I had the opportunity to learn something today. Thanks @jmarrec :)